### PR TITLE
Add tests for check_user action, which also covers the proxy use case

### PR DESCRIPTION
**Pull request by bgeuken**

### CODE REVIEW

File: src/api/test/functional/webui/webui_controller_test.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+class Webui::WebuiControllerTest < Webui::IntegrationTest
+
+  setup do
+    @before = CONFIG['proxy_auth_mode']
+    # Fake proxy mode
+    CONFIG['proxy_auth_mode'] = :on
+  end
+
+  teardown do
+    CONFIG['proxy_auth_mode'] = :off
```
Comment (mdeniz) [inline src/api/test/functional/webui/webui_controller_test.rb]: Shouldn't be ```CONFIG['proxy_auth_mode'] = @before``` ??

Comment (bgeuken) [inline src/api/test/functional/webui/webui_controller_test.rb]: Oups. You are right.
And actually I could use stubs here. I'll rebase in a bit. Thanks!

**Comment by bgeuken** [comment]
@mdeniz Mind to have a second look?

**Comment by mdeniz** [comment]
The big difference is that this is using the UI and javascript. If you find it useful LGTM :+1:

**Comment by bgeuken** [comment]
AFAIK the use_js call was only needed for find('#link-to-user-home') to work.

**Comment by hennevogel** [comment]
`test_access_as_new_user` is testing user creation too. I would like to have one place where we test this. not two please

**Comment by bgeuken** [comment]
I prefer this one since, it doesn't require me to change the configuration. And I find it more readable. But that might be because I wrote it^^

**Comment by adrianschroeter** [comment]
this solution is fine for me, but we need to drop the proxy test suite run in obs-server.spec and also for travis again if we drop the other test case.

**Comment by bgeuken** [comment]
Approved in person by Henne

=> merging

**Pull request merged**